api/domain_handlers.go

```diff
@@ -381,3 +381,55 @@ func (api API) Remove(w http.ResponseWriter, r *http.Request) {
 
 	writeJSONOrBust(w, issues)
 }
+
+// Example: GET /ineligible?
+func (api API) Ineligible(w http.ResponseWriter, r *http.Request) {
+
+	var ineligibleDomains []database.IneligibleDomainState
+	var deleteEligibleDomains []string
+
+	domains, err := api.database.AllDomainStates()
+	if err != nil {
+		msg := fmt.Sprintf("Internal error: could not retrieve domains. (%s)\n", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	}
+	// Store ineligible domains in slice
+	for _, d := range domains {
+		_, issues := hstspreload.EligibleDomain(d.Name, d.Policy)
+
+		if len(issues.Errors) > 0 {
+			ineligibleDomains = append(ineligibleDomains, database.IneligibleDomainState{
+				Name: d.Name,
+				Scans: []database.Scan{
+					{
+						ScanTime: time.Now(),
+						Issues:   []hstspreload.Issues{issues},
```

Comment:
(I missed this when reviewing a previous PR.) I think that the `Issues` field in the `Scan` struct should be of type `hstspreload.Issues`, not `[]hstspreload.Issues`. The `Issues` type already encapsulates a set of errors and warnings, and as demonstrated by the usage here (the output from `hstspreload.EligibleDomain`), there will be only one `Issues` for each scan.